ci/autotests.py

```diff
@@ -81,12 +81,14 @@ def run(scenarios=None, send_to_testrail=False, fail_on_failed_scenario=False, o
                 raise AttributeError('Attribute `{0}` does not exists as status in TestrailResult'.format(module_result['status']))
             # add test to results & also remove possible EXCLUDE_FLAGS on test name
             results[test.replace(AutoTests.EXCLUDE_FLAG, '')] = module_result
+            if send_to_testrail:
+                logger.info("Start pushing test {0} to testrail.".format(test.split('.')[-1]))
+                plan_url = AutoTests.push_to_testrail({test: module_result}, only_add_given_cases=only_add_given_results)
+                results[test.replace(AutoTests.EXCLUDE_FLAG, '')]['plan_url'] = plan_url
```

Comment:
Some other stuff needs to happen. The push to testrail creates a plan now for every test. This is not what we want.
Instead:
Before the loop:

- Verify that tests are specified
- Create a plan if tests are specified
- Link suite to the plan
- Add test cases to the plan
- Add results one at the time